File: .github/scripts/src/Automation/WorkflowState.php
<?php

declare(strict_types=1);

namespace DockerBase\Automation;

enum WorkflowState: string
{
    case Missing = 'missing';
    case Pending = 'pending';
    case Succeeded = 'succeeded';
    case Failed = 'failed';
    case Cancelled = 'cancelled';
    case TimedOut = 'timed_out';
}

File: .github/scripts/automation/workflow_state.py
from enum import Enum


class WorkflowState(Enum):
    """A closed set of states used by polling orchestration."""

    MISSING = 'missing'
    PENDING = 'pending'
    SUCCEEDED = 'succeeded'
    FAILED = 'failed'
    CANCELLED = 'cancelled'
    TIMED_OUT = 'timed_out'
